src/main/java/org/apache/maven/buildcache/checksum/DigestUtils.java

```diff
@@ -126,6 +126,10 @@ public static DigestItem dependency(HashChecksum checksum, String key, String ha
         return item("dependency", key, checksum.update(hash));
     }
 
+    public static DigestItem pluginDependency(HashChecksum checksum, String key, String hash) {
```

Comment:
Sorry, I forgot to mention that there is a `CacheDiff` tool. It will also be good to add the plugin dependencies to the diff report.